packages/graphql-migrations/src/connector/read.ts

```diff
@@ -83,20 +82,28 @@ class Reader {
       const columnComments = await getColumnComments(this.knex, tableName, this.schemaName)
       const columnInfo: { [key: string]: Knex.ColumnInfo } = await this.knex(tableName)
         .withSchema(this.schemaName)
-        .columnInfo() as any
+        .columnInfo() as any;
+      // Primary key
+      const primaries = await getPrimaryKey(this.knex, tableName, this.schemaName)
+      
       // eslint-disable-next-line no-restricted-syntax
       for (const key in columnInfo) {
         if (columnInfo[key]) {
           const columnName = this.getColumnName(key)
           if (!columnName) { continue }
-          const info = columnInfo[key]
-          const foreign = foreignKeys.find((k: any) => k.column === key)
+          const info = columnInfo[key];
+          const {args, type} = getTypeAlias(info.type, info.maxLength);
+          const foreign = foreignKeys.find((k: any) => k.column === key);
+          const isPrimaryKey  = primaries.some((primary:  {column: string, indexName: string }) => primary.column === columnName);
           const column: TableColumn = {
             name: columnName,
+            isPrimaryKey,
+            autoIncrementable: type === "increments", // TODO
             comment: this.getComment(columnComments, key),
             annotations: {},
```

Comment:
So is `args and `type` moved out of annotations to a top-level property in `TableColumn` now?

Reply:
They were always top level property because of the destructing that happened in this line https://github.com/aerogear/graphback/pull/1421/files#diff-2038add530cf853b61f15d1ee2036174L98

I just extracted the info caused I needed it in line 101.